File: HW/HW11/Image.py
class Image:
    def floodFill(self, image, sr, sc, color):
        if not image or image[sr][sc] == color or len(image) == 0 or len(image[0]) == 0:
            return image

        self.graph(image, sr, sc, image[sr][sc], color)
        return image

    def graph(self, image, row, col, oldColor, newColor):
        if row < 0 or row >= len(image) or col < 0 or col >= len(image[0]) or image[row][col] != oldColor:
            return
        image[row][col] = newColor

        self.graph(image, row + 1, col, oldColor, newColor)
        self.graph(image, row, col + 1, oldColor, newColor)
        self.graph(image, row - 1, col, oldColor, newColor)
        self.graph(image, row, col - 1, oldColor, newColor)

File: HW/HW11/Image.java
public class Image {
    public int[][] floodFill(int[][] image, int sr, int sc, int color)
    {
        if (image == null||image[sr][sc] == color || image.length == 0 || image[0].length == 0)
        {
            return image;
        }

        graph(image, sr, sc, image[sr][sc], color);
        return image;
    }

    private void graph(int[][] image, int row, int col, int oldColor, int newColor)
    {
        if (row < 0 || row >= image.length || col < 0 || col >= image[0].length || image[row][col] != oldColor) {
            return;
        }
        image[row][col] = newColor;

        graph(image, row + 1, col, oldColor, newColor);
        graph(image, row, col + 1, oldColor, newColor);
        graph(image, row - 1, col, oldColor, newColor);
        graph(image, row, col - 1, oldColor, newColor);
    }
}

